app/src/main/java/org/thoughtcrime/securesms/components/subsampling/AttachmentRegionDecoder.java

```diff
@@ -0,0 +1,81 @@
+package org.thoughtcrime.securesms.components.subsampling;
+
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.BitmapRegionDecoder;
+import android.graphics.Point;
+import android.graphics.Rect;
+import android.net.Uri;
+
+import com.davemorrissey.labs.subscaleview.decoder.ImageRegionDecoder;
+import com.davemorrissey.labs.subscaleview.decoder.SkiaImageRegionDecoder;
+
+import org.signal.core.util.logging.Log;
+import org.thoughtcrime.securesms.mms.PartAuthority;
+
+import java.io.InputStream;
+
+public class AttachmentRegionDecoder implements ImageRegionDecoder {
+
+  private static final String TAG = Log.tag(AttachmentRegionDecoder.class);
+
+  private SkiaImageRegionDecoder passthrough;
+
+  private BitmapRegionDecoder bitmapRegionDecoder;
+
+  @Override
+  public Point init(Context context, Uri uri) throws Exception {
+    Log.d(TAG, "Init!");
+    if (!PartAuthority.isLocalUri(uri)) {
+      passthrough = new SkiaImageRegionDecoder();
+      return passthrough.init(context, uri);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;context&#39; (index: 0 | type: Context). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;uri&#39;. The same parameter type is validated here for example: app/src/flipper/java/org/thoughtcrime/securesms/database/FlipperSqlCipherAdapter.java:40. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).